sql-plugin/src/main/spark350/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

```diff
@@ -21,13 +21,29 @@ package com.nvidia.spark.rapids.shims
 
 import com.nvidia.spark.rapids._
 
-import org.apache.spark.sql.catalyst.expressions.{Expression, PythonUDAF}
+import org.apache.spark.sql.catalyst.expressions.{Expression, PythonUDAF, ToPrettyString}
 import org.apache.spark.sql.rapids.execution.python.GpuPythonUDAF
+import org.apache.spark.sql.types.BinaryType
 
 object SparkShimImpl extends Spark340PlusShims {
 
   override def getExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = {
     val shimExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = Seq(
+      GpuOverrides.expr[ToPrettyString]("An internal expressions which is used to " +
+        "generate pretty string for all kinds of values",
+        ExprChecks.unaryProject(TypeSig.STRING, TypeSig.STRING, TypeSig.all, TypeSig.all),
+        (a, conf, p, r) => new UnaryExprMeta[ToPrettyString](a, conf, p, r) {
+          override def tagExprForGpu(): Unit = {
+            if (a.asInstanceOf[ToPrettyString].child.dataType == BinaryType) {
```

Comment:
Why do we say that we support all types as input in the TypeSig when we don't support Binary?  What about an ARRAY[Binary] Do we support that? This is not checking for that when the TypeSig would. What about user defined types?